context/src/main/java/io/grpc/Context.java

```diff
@@ -881,8 +881,10 @@ public String toString() {
      * Implements {@link io.grpc.Context#attach}.
      *
      * @param toAttach the context to be attached
+     * @return A {@link Context} that should be passed back into {@link #detach(Context, Context)}
+     *        as the {@code toRestore} parameter.
      */
-    public abstract void attach(Context toAttach);
+    public abstract Context attach(Context toAttach);
```

Comment:
IIRC this is a binary-incompatible change (you may want to verify it). Although the storage API is experimental thus allows us to break, our practice has been that we will provide smooth transition as long as it doesn't take much effort.

If this is binary-incompatible, a trick I usually use is:
```java
@Deprecated
public void attach(Context toAttach) {
  throw new UnsupportedOperationException("Deprecated. Do not call.");
}

public Context doAttach(Context toAttach) {
  attach(toAttach);
  return Context.current();
}
```

After one release, delete `attach()` and make `doAttach()` abstract.